app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt

```diff
@@ -1,5 +1,246 @@
+                        zm x: Czy = prawda
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions)`() {
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        czynność g() -> Liczba {
+                            zwróć 17
+                        }
+                        czynność g() -> Liczba {
+                            zwróć 18
+                        }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions with different return types)`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Liczba {
+                        czynność g() -> Czy {
+                            zwróć prawda
+                        }
+                        czynność g() -> Nic {
+                            zakończ
+                        }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions with different signatures)`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Liczba {
+                        czynność g(a: Liczba) { }
+                        czynność g(a: Czy) { }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    // Assuming no functional features.
+
+    @Ignore
+    @Test
+    fun `test calling variables`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Czy {
+                        zm x: Czy = prawda
+                        zwróć x()
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.VariableIsNotCallable::class
+        )
+    }
+
+    // I'm not 100% sure if the next two tests make sense
+    @Ignore
+    @Test
+    fun `test using functions in assignment`() {
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        czynność g() { }
+                        zm x: Liczba = x
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.FunctionIsNotVariable::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test using functions in conditions`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Czy {
+                        czynność g() { }
+                        zwróć (g == 17)
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.FunctionIsNotVariable::class
```

Comment:
This test is probably redundant - the previous one already tests if `g` is a correct expression